src/azul/project/hca/transformers.py

```diff
@@ -369,16 +435,27 @@ def transform(self,
                             manifest=manifest,
                             metadata_files=metadata_files)
         project = self._get_project(bundle)
-        for specimen in bundle.specimens:
+        samples: MutableMapping[str, Sample] = dict()
+        for si in bundle.sequencing_input:
+            SampleTransformer.get_ancestor_samples(samples, si)
+        for sample in samples.values():
             visitor = TransformerVisitor()
-            specimen.accept(visitor)
-            specimen.ancestors(visitor)
-            contents = dict(specimens=[_specimen_dict(specimen, visitor.visited_entities)],
+            sample.accept(visitor)
+            sample.ancestors(visitor)
+            contents = dict(samples=[_sample_dict(sample)],
+                            specimens=[_specimen_dict(s) for s in visitor.specimens.values()],
                             cell_suspensions=[_cell_suspension_dict(cs) for cs in visitor.cell_suspensions.values()],
+                            cell_lines=[_cell_line_dict(cl) for cl in visitor.cell_lines.values()],
+                            donors=[_donor_dict(d) for d in visitor.donors.values()],
+                            organoids=[_organoid_dict(o) for o in visitor.organoids.values()],
                             files=[_file_dict(f) for f in visitor.files.values()],
                             protocols=[_protocol_dict(pl) for pl in visitor.protocols.values()],
                             projects=[_project_dict(project)])
-            yield self._contribution(bundle, contents, specimen)
+            entity_types = {'cell_line': 'cell_lines', 'organoid': 'organoids', 'specimen_from_organism': 'specimens'}
+            sample_entity_type = one(contents['samples'])['sample_entity_type']
```

Comment:
```suggestion
            sample_type = one(contents['samples'])['sample_type']
```

My intent behind the comment from yesterday was that there would be two variables, one called `sample_entity_type` and the other `sample_type`. The former would contain `specimens` and the latter `specimen_from_organism`.

However, I changed my mind. See my comment on `_sample_dict`.